### PR TITLE
Prevent commands which take Tasks IDs taking Job IDs.

### LINKED ISSUE #6125: commands: traceback when specifying jobs
Scheduler log traceback appears when jobs are provided as task arguments, e.g:

```
cylc release workflow//<cycle>/<task>/01
```

### OP:

Running against any old workflow:

`cylc <workflow>//2397/bar/NN/` (got by pressing tab and letting the autocomplete go to it).

Gives the `command queued` output in the dispatching terminal, but the following in the log:

```
2024-06-06T16:03:50+01:00 INFO - Command "set" received. ID=89921d9f-9447-4cb4-acce-5fc9f972076e
    set(flow=['all'], flow_wait=False, outputs=[], prerequisites=[], tasks=['2397/bar/NN'])
2024-06-06T16:03:50+01:00 ERROR - Traceback (most recent call last):
      File "/net/home/h02/tpilling/metomi/cylc-flow/cylc/flow/scheduler.py", line 948, in process_command_queue
        n_warnings = await cmd.__anext__()
      File "/net/home/h02/tpilling/metomi/cylc-flow/cylc/flow/commands.py", line 155, in set_prereqs_and_outputs
        yield schd.pool.set_prereqs_and_outputs(
      File "/net/home/h02/tpilling/metomi/cylc-flow/cylc/flow/task_pool.py", line 1887, in set_prereqs_and_outputs
        itasks, future_tasks, unmatched = self.filter_task_proxies(
      File "/net/home/h02/tpilling/metomi/cylc-flow/cylc/flow/task_pool.py", line 2285, in filter_task_proxies
        matched, unmatched = filter_ids(
      File "/net/home/h02/tpilling/metomi/cylc-flow/cylc/flow/id_match.py", line 207, in filter_ids
        raise NotImplementedError
    NotImplementedError
    
2024-06-06T16:03:50+01:00 ERROR - Command "set" failed. ID=89921d9f-9447-4cb4-acce-5fc9f972076e
```

_Originally posted by @wxtim in https://github.com/cylc/cylc-flow/issues/6112#issuecomment-2152774732_
 
 
 Similarly 

`cylc hold --after 239413 <workflow>` is accepted but later causes traceback.

### PR DESCRIPTION
**Pull request by wxtim**
Doing so can cause the scheduler to crash.

Branched from #6112 - drafted until that is in.
Closes #6125 

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] Whilst this is technically a bug fix, I'm using a feature only available on a branch targetting master to fix it.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
Suggest raising this as a separate PR when time permits (not merging into an 8.3.0 PR).

Suggestion from the issue is to translate IDs into tokens then pass them down the chain in order to avoid re-tokenising later.

https://github.com/cylc/cylc-flow/issues/6125#issuecomment-2152857066

File: cylc/flow/command_validation.py [hunk 1]
```diff
@@ -228,3 +228,40 @@ def consistency(
     """
     if outputs and prereqs:
         raise InputError("Use --prerequisite or --output, not both.")
+
+
+def is_tasks(tasks: List[str]):
+    """All tasks in a list of tasks are task ID's
+    without trailing job ID.
+
```
Comment (wxtim) [inline cylc/flow/command_validation.py:236]:
```suggestion

```

File: cylc/flow/command_validation.py [hunk 2]
```diff
@@ -228,3 +228,39 @@ def consistency(
     """
     if outputs and prereqs:
         raise InputError("Use --prerequisite or --output, not both.")
+
+
+def is_tasks(tasks: List[str]):
+    """All tasks in a list of tasks are task ID's
+    without trailing job ID.
+    Examples:
```
Comment (wxtim) [inline cylc/flow/command_validation.py:236]:
```suggestion

    Examples:
```

**Review by oliver-sanders — COMMENTED** [review]
Small stuff:

**Review by oliver-sanders — COMMENTED** [review]
LGTM, tested as working, but there are a couple more commands to extend this to:

* set
* stop (stop tasks)
* hold (hold tasks)

**Comment by wxtim** [comment]
> LGTM, tested as working, but there are a couple more commands to extend this to:
> 
>     * set 
>     * stop (stop tasks)

Added these

>     * hold (hold tasks)

Pretty sure this was already added.

**Review by oliver-sanders — APPROVED** [review]

File: changes.d/6130.fix.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Prevent commands taking a list of task ID's from submitting job ids.
```
Comment (MetRonnie) [inline changes.d/6130.fix.md:1]:
```suggestion
Prevent commands accepting job IDs where it doesn't make sense.
```

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**